src/gfn/modules.py

```diff
@@ -212,10 +193,7 @@ def forward(self, input: States | torch.Tensor) -> torch.Tensor:
         if out.shape[-1] != 1:
             out = self.reduction_fxn(out, -1)
 
-        if not self._output_dim_is_checked:
-            self.check_output_dim(out)
-            self._output_dim_is_checked = True
-
+        assert out.shape[-1] == 1
```

Comment:
This also seems like a much harder constraint.

Reply:
Why? The `expected_output_dim` is 1 in this class, so it seems the same to me (actually softer as I don't check the dtype).